src/Microsoft.ML.KMeansClustering/KMeansCatalog.cs

```diff
@@ -26,7 +26,7 @@ public static class KMeansClusteringExtensions
         ///  [!code-csharp[KMeans](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Clustering/KMeans.cs)]
         /// ]]></format>
         /// </example>
-        public static KMeansPlusPlusTrainer KMeans(this ClusteringCatalog.ClusteringTrainers catalog,
+        public static KMeansPlusPlusTrainer KMeansPlusPlus(this ClusteringCatalog.ClusteringTrainers catalog,
```

Comment:
I think we should be consistent everywhere about this name. If we really intend for this to be `KMeansPlusPlus`, then we should update all the places that just use `KMeans`:

The name of the above class:
`public static class KMeansClusteringExtensions`

The name of the assembly:
`Microsoft.ML.KMeansClustering`

Is there any confusion about just using the name `KMeans`? Are there other `KMeans` algorithms besides `KMeans++`? It feels like we should be fine just using `KMeans`, but I'll leave it up to the experts.

Reply:
[Kmeans++](https://en.wikipedia.org/wiki/K-means%2B%2B) is a slightly modified version of [Kmeans] ([https://en.wikipedia.org/wiki/K-means_clustering](https://en.wikipedia.org/wiki/K-means_clustering)) that has some smarts for choosing the initial cluster centers. The trainer estimator we have currently  implements  Kmeans++ 

For the specific  trainer implementation we have currently, we are using `KMeansPlusPlus`  for the MLContext name and `KMeansPlusPlusTrainer`  the Class name

I can envision that in the future one might even want to implement the vanilla Kmeans  algorithm itself.  As such to me it makes sense  to keep  the  name of  the static class  and assembly as just `KMeans` .   

---
In reply to: [264707009](https://github.com/dotnet/machinelearning/pull/2903#discussion_r264707009) [](ancestors = 264707009)

Reply:
My feeling is that no one will add Kmeans as long as Kmeans++ exists. As you mentioned, Kmeans++ is a member of Kmeans family. Why can't we call it Kmeans? If someone want to implement the original Kmeans, it can be called NaiveKmeans.

Reply:
Sure. That makes sense.  Will rename it to `KMeans`  to keep things uniform

Reply:
Thanks. I also noticed there is a property  `InitializationAlgorithm`  to specify the initialization  mechanism 

So yeah. It should be just `KMeans`

---
In reply to: [264770186](https://github.com/dotnet/machinelearning/pull/2903#discussion_r264770186) [](ancestors = 264770186)

Reply:
fixed.  We should indeed call it `KMeans` 

---
In reply to: [264757754](https://github.com/dotnet/machinelearning/pull/2903#discussion_r264757754) [](ancestors = 264757754,264707009)